src/test/java/org/candlepin/subscriptions/resource/TallyResourceTest.java

```diff
@@ -67,24 +71,31 @@
 @SpringBootTest
 @ActiveProfiles({"api", "test"})
 @WithMockRedHatPrincipal("123456")
-@Import(FixedClockConfiguration.class)
+@Import(TestClockConfiguration.class)
 class TallyResourceTest {
+  public static final Instant MID_MONTH_INSTANT =
+      LocalDateTime.of(2023, 7, 15, 13, 0, 0).toInstant(ZoneOffset.UTC);
+  public static final OffsetDateTime TEST_DATE =
+      OffsetDateTime.ofInstant(MID_MONTH_INSTANT, ZoneOffset.UTC);
 
   public static final ProductId RHEL_PRODUCT_ID = ProductId.RHEL;
-  public static final String INVALID_PRODUCT_ID_VALUE = "bad_product";
+
   private final OffsetDateTime min = OffsetDateTime.now().minusDays(4);
   private final OffsetDateTime max = OffsetDateTime.now().plusDays(4);
-  private static ApplicationClock CLOCK = new FixedClockConfiguration().fixedClock();
 
   @MockBean TallySnapshotRepository repository;
   @MockBean BillableUsageRemittanceRepository remittanceRepository;
   @MockBean PageLinkCreator pageLinkCreator;
   @MockBean AccountConfigRepository accountConfigRepository;
   @Autowired TallyResource resource;
 
+  @Autowired ApplicationClock applicationClock;
+
   @BeforeEach
   public void setupTests() {
     when(accountConfigRepository.existsByOrgId("owner123456")).thenReturn(true);
+    var testClock = (TestClock) applicationClock.getClock();
+    testClock.setInstant(MID_MONTH_INSTANT);
```

Comment:
Doesn't this pollute state for other tests? I don't mind having an adjustable clock, but then injecting it to be shared seems problematic.

Reply:
I thought that the application context was brought up on a per-test class basis.  Your point is well taken though.  I will see if I can make some adjustments here.

Reply:
See https://docs.spring.io/spring-framework/reference/testing/testcontext-framework/ctx-management/caching.html